ts/main.ts

```diff
@@ -48,19 +47,48 @@ $entryForm.addEventListener('submit', function (event: Event) {
   };
   // stores the form's input values in a new object
   // assigns an `entryId` property to the new object, taken from the `nextEntryId` property of the data model.
-  data.nextEntryId++;
-  // increments the `nextEntryId` property of the data model so if another entry submitted later, it will receive diff `entryId`.
-  data.entries.unshift(newObj);
-  // adds the new object to the beginning of the data model's array of entries.
-  writeData();
-  const renderedEntry = renderEntry(newObj);
-  $entriesList.prepend(renderedEntry);
-  viewSwap('entries');
-  toggleNoEntries();
-  $previewImg.setAttribute('src', '../images/placeholder-image-square.jpg');
-  // resets the preview image's `src` attribute back to the placeholder image.
-  $entryForm.reset();
-  // resets form
+  if (data.editing === null) {
+    data.nextEntryId++;
+    // increments the `nextEntryId` property of the data model so if another entry submitted later, it will receive diff `entryId`.
+    data.entries.unshift(newObj);
+    // adds the new object to the beginning of the data model's array of entries.
+    writeData();
+    const renderedEntry = renderEntry(newObj);
+    $entriesList.prepend(renderedEntry);
+    viewSwap('entries');
+    toggleNoEntries();
```

Comment:
You run your `writeData`, `viewSwap`, and `$entryForm.reset()` multiple times in this function but you run it in different conditions. Instead of running them in the conditions, run them _after_ your conditional code since we _always_ want to write the data, change views, and reset the form on form submission.